packages/cli/src/commands/pull.ts

```diff
@@ -77,32 +81,93 @@ export default async function main(client: Client) {
 
     if (link.status === 'not_linked') {
       // User aborted project linking questions
-      return 0;
+      return { linkResultCode: 0 };
     }
   }
 
   if (link.status === 'error') {
-    return link.exitCode;
+    return { linkResultCode: link.exitCode };
   }
 
-  const { project, org } = link;
+  return { link, linkResultCode: 0 };
+}
 
-  client.config.currentTeam = org.type === 'team' ? org.id : undefined;
+async function pullAllEnvFiles(
+  envFileRoot: string,
+  client: Client,
+  project: Project,
+  argv: any,
+  cwd: string
+) {
+  const devEnvFile = `${envFileRoot}.development.local`;
+  const pullDevPromise = envPull(
+    client,
+    project,
+    ProjectEnvTarget.Development,
+    argv,
+    [join(cwd, devEnvFile)],
+    client.output
+  );
 
-  const result = await pull(
+  const previewEnvFile = `${envFileRoot}.preview.local`;
+  const pullPreviewPromise = envPull(
     client,
     project,
+    ProjectEnvTarget.Preview,
     argv,
-    [join(cwd, env)],
+    [join(cwd, previewEnvFile)],
     client.output
   );
-  if (result !== 0) {
-    // an error happened
-    return result;
+
+  const prodEnvFile = `${envFileRoot}.production.local`;
+  const pullProdPromise = envPull(
+    client,
+    project,
+    ProjectEnvTarget.Production,
+    argv,
+    [join(cwd, prodEnvFile)],
+    client.output
+  );
+
+  const [pullDevResultCode, pullPreviewResultCode, pullProdResultCode] =
+    await Promise.all([pullDevPromise, pullPreviewPromise, pullProdPromise]);
+
+  return pullDevResultCode || pullPreviewResultCode || pullProdResultCode || 0;
+}
+
+export default async function main(client: Client) {
+  const { argv, argResultCode } = parseArgs(client);
+  if (argResultCode !== 0) {
+    return argResultCode;
+  }
+  if (!argv) {
+    throw new Error('argResultCode was 0, but `argv` did not exist');
+  }
```

Comment:
I was trying to find a way to type this function such that it either has a non-zero exit code OR has an `argResultCode` value. That way, I wouldn't have this extra check for `argResultCode` missing that will essentially never happen. (Same goes for the other result code functions below.)

I could rely on throwing an exception in non-zero cases, making the function signature simpler to manage, but then the consuming code has to wrap in try-catch. We don't get away from extra code that's noise to the reader most of the time.

If anyone can think of a better way to type these, let me know.